imagetool/src/main/java/com/oracle/weblogic/imagetool/util/HttpUtil.java

```diff
@@ -120,25 +116,64 @@ public static HttpClient getOraClient(String userId, String password) {
         config.setCookieSpec(CookieSpecs.STANDARD);
 
         CookieStore cookieStore = new BasicCookieStore();
-        CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
-
-        if (userId != null && password != null) {
-            BasicClientCookie cc = new BasicClientCookie("oraclelicense", "a");
-            cc.setDomain("edelivery.oracle.com");
-            cookieStore.addCookie(cc);
-            credentialsProvider.setCredentials(AuthScope.ANY, new UsernamePasswordCredentials(
-                    userId, password));
+
+        String proxyHost = System.getProperty("https.proxyHost");
+        String proxyPort  = System.getProperty("https.proxyPort");
+        HttpClient result;
+        if (proxyHost != null) {
+            result = HttpClientBuilder.create()
+                .setDefaultRequestConfig(config.build())
+                .setRetryHandler(retryHandler())
+                .setProxy(new HttpHost(proxyHost, Integer.parseInt(proxyPort)))
+                .setUserAgent("Wget/1.10")
+                .setDefaultCookieStore(cookieStore).useSystemProperties()
+                .build();
+
+        } else {
+            result = HttpClientBuilder.create()
+                .setDefaultRequestConfig(config.build())
+                .setRetryHandler(retryHandler())
+                .setUserAgent("Wget/1.10")
+                .setDefaultCookieStore(cookieStore).useSystemProperties()
+                .build();
+
         }
-        HttpClient result = HttpClientBuilder.create()
-            .setDefaultRequestConfig(config.build())
-            .setRetryHandler(retryHandler())
-            .setUserAgent("Wget/1.10")
-            .setDefaultCookieStore(cookieStore).useSystemProperties()
-            .setDefaultCredentialsProvider(credentialsProvider).build();
         logger.exiting();
         return result;
     }
 
+    /**
+     * Return a Executor for http access.
+     * @param supportUserName  oracle support username
+     * @param supportPassword oracle support password
+     * @return Executor
+     */
+
+    public static Executor getHttpExecutor(String supportUserName, String supportPassword) {
+
+        String proxyUser = System.getProperty("https.proxyUser");
+        String proxyPassword = System.getProperty("https.proxyPassword");
+        String proxyHost = System.getProperty("https.proxyHost");
+        String proxyPort  = System.getProperty("https.proxyPort");
+        Executor executor =  Executor.newInstance(getOraClient(supportUserName, supportPassword));
+        if (proxyHost != null) {
+            if (proxyPassword != null) {
+                executor
+                    .auth(new HttpHost(proxyHost, Integer.parseInt(proxyPort)), proxyUser, proxyPassword)
```

Comment:
Similar comment to the above, the if/else branches with duplicated content makes this harder to read and understand. It would be better if we consolidated common code.

Reply:
The auth method takes three parameters, I don't think we should set null proxy and then pass user credentials.